components/ILIAS/Certificate/classes/Overview/CertificateOverviewTable.php

```diff
@@ -101,19 +102,33 @@ public function getRows(
         ?array $additional_parameters
     ): Generator {
         /**
-         * @var array{certificate_id: null|string, issue_date: null|DateTime, object: null|string, owner: null|string} $filter_data
+         * @var array{certificate_id: null|string, issue_date: null|DateTimeImmutable, object: null|string, owner: null|string} $filter_data
          */
         $ui_filter_data = $this->ui_service->filter()->getData($this->filter);
         [$order_field, $order_direction] = $order->join([], fn($ret, $key, $value) => [$key, $value]);
 
         if (isset($ui_filter_data['issue_date']) && $ui_filter_data['issue_date'] !== '') {
             try {
-                $ui_filter_data['issue_date'] = new DateTime($ui_filter_data['issue_date']);
-            } catch (Exception) {
-                $ui_filter_data['issue_date'] = null;
+                $from = new DateTimeImmutable($ui_filter_data['issue_date'][0], $this->user_timezone);
+            } catch (Throwable) {
+                $from = null;
             }
+
+            try {
+                $to = new DateTimeImmutable($ui_filter_data['issue_date'][1], $this->user_timezone);
+            } catch (Throwable) {
+                $to = null;
+            }
+
+            $ui_filter_data['issue_date'] = [
+                'from' => $from,
+                'to' => $to
+            ];
         } else {
-            $ui_filter_data['issue_date'] = null;
+            $ui_filter_data['issue_date'] = [
+                'from' => null,
+                'to' => null
+            ];
         }
```

Comment:
This looks similar to the code in `getTotalRowCount`, please extract this to one method.